### DISCUSSION TITLE
in V11, DataGrid.AlternatingRowBackground missing

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#12084] [Q&A] [answered]
Hi,

What's the alternative of DataGrid.AlternatingRowBackground in V11 now?

I am just migrating to V11 and this is the only issue left.

Joachim

**Comment by timunie** [comment]
iirc the nth-child selector can now be used on DataGridRow-Styles https://docs.avaloniaui.net/docs/next/reference/styles/style-selector-syntax#by-child-position-formula

**Reply by drei97** [reply]
And how is this working in detail?

I set up

```
    <Style Selector="DataGrid:nth-child(odd)">
        <Setter Property="Background" Value="Red"/>
    </Style>

```

But there is no alternating coloring

```
           <DataGrid x:Name="datagrid"
                      ItemsSource="{Binding Data}"
                      AutoGenerateColumns="False"
                      IsReadOnly="True"
                      SelectedItem="{Binding SelectedItem}">
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/68755045/05571b5d-0d9e-4978-ba73-5348a71b590d)

It seems, something is missing

**Reply by timunie** [reply, marked as answer]
DataGridRow must be the selector I guess.

**Reply by drei97** [reply]
You are right.

Thnx
